bin/journey-test.sh

```diff
@@ -233,15 +233,15 @@ solve_all_exercises() {
 
     pushd ${exercism_exercises_dir}/${TRACK}/${exercise}
     # Check that tests compile before we strip @Ignore annotations
-    "$EXECPATH"/gradlew compileTestJava
+    "$EXECPATH"/gradlew compileTestJava --configure-on-demand --parallel
```

Comment:
The [gradle docs] say this(https://github.com/gradle/gradle/blob/master/design-docs/configure-on-demand.md).

> Gradle evaluates all projects even though it might not be necessary. Imagine a 300-module build and someone just wants to run a single task from a single subproject. It would be nice if Gradle only evaluated the projects that are necessary for the selected task to run.